src/pages/learn/index.mdx

```diff
@@ -1,42 +1,46 @@
+import { Callout } from "nextra/components"
+
 # Introduction to GraphQL
 
-> Learn about GraphQL, how it works, and how to use it. Looking for documentation on how to build a GraphQL service?
-> There are libraries to help you implement GraphQL in [many different languages](/code). For an in-depth learning experience
-> with practical tutorials, see [the available training courses](/community/resources/training-courses).
+<p className="learn-subtitle">Learn about GraphQL, how it works, and how to use it</p>
+
+GraphQL is a query language for your API, and a server-side runtime for executing queries using a type system you define for your data. The [GraphQL specification](https://spec.graphql.org/) was open-sourced in 2015 and has since been implemented in a variety of programming languages. GraphQL also isn't tied to any specific database or storage engine and is instead backed by your existing code and data.
 
-GraphQL is a query language for your API, and a server-side runtime for executing queries using a type system you define for your
-data. GraphQL isn't tied to any specific database or storage engine and is instead backed by your existing code and data.
+## Describe your API with a type system
 
-A GraphQL service is created by defining types and fields on those types, then providing functions for each field on each type.
-For example, a GraphQL service that tells you who the logged in user is (`me`) as well as that user's name might look
-like this:
+A GraphQL service is created by [defining types and their fields](/learn/schema/), and then writing functions that [provide data for each field](/learn/execution/). For example, a GraphQL service that tells you the name of a logged-in user might look like this:
 
 ```graphql
 type Query {
   me: User
 }
 
 type User {
-  id: ID
   name: String
 }
 ```
 
 Along with functions for each field on each type:
 
 ```js
-function Query_me(request) {
-  return request.auth.user
+Query: {
```

Comment:
This is no longer standalone JS code (`Query` here would be interpreted as a [label](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label) which could be very confusing for readers new to JavaScript), and it ties into the `{[typeName]: {[fieldName]: resolver}}` pattern common in various tools across the JavaScript ecosystem but not actually supported by graphql.js natively and also not as portable to other programming languages.

`function Query_me(request)` has the advantage of being fairly generic pseudocode even if the reader does not know JavaScript.

Perhaps adding a comment before the function would be a suitable alternative solution:

```ts
// The resolver for the `me` field on the `Query` type, aka `Query.me`:
function Query_me(request) {
```

and later:

```ts
// The resolver for the `User.name` field:
function User_name(user) {
```

what do you think?

Reply:
I see what you mean with the label confusion. My intent here was standardize these examples with the JS-ish examples that were already on the Execution page. But maybe a `function {Type}_{field}` () {}` syntax would be best to use throughout the entire guide instead? Let me know what you think and I can adjust them.

Reply:
Urgh, yes, how inconsistent! I do like the structure of typeName/fieldName for resolver clarity but the invalid JS code is not helpful. Out of interest I fed it into ChatGPT to see what it made of it:

Converting the invalid JS code to Python:

https://chatgpt.com/share/671a2803-2338-8012-aaee-c7eb490b72ce (doesn't look useful)

Converting expanded valid JS code to Python:

https://chatgpt.com/share/671a2843-3f08-8012-83d7-8769c471ecd9 (looks much more reasonable to me).

Disclaimer: I've not written Python for ~8 years.

I think I'd stick with functions like you suggest above on this page, and then on the execution page if it makes sense to do the typeName/fieldName structure, explicitly introduce it (e.g. "for organizational reasons, you may wish to keep your resolvers in a structured object, for example ...")

Reply:
Sounds good. I updated the functions as discussed and moved the libraries callout in the latest commit so this should resolve all of the feedback for this page.